activerecord/lib/active_record/test_fixtures.rb

```diff
@@ -111,12 +111,11 @@ def setup_fixtures(config = ActiveRecord::Base)
       @fixture_connections = []
       @@already_loaded_fixtures ||= {}
       @connection_subscriber = nil
+      @legacy_saved_pool_configs = Hash.new { |hash, key| hash[key] = {} }
```

Comment:
While we're making changes in here can you condition these or make one instance var? I don't want the legacy one available when we're not in legacy mode.

Reply:
Done! I kept the variables separate but made sure they were only defined and used on their respective connection handling logic since it's more explicit and would allow us to just remove the legacy connection block once we remove that.

Reply:
@eileencodes I had to revert the change to condition these variables since it looks like the tests call out a setup and teardown with different values causing the variable to load values for legacy and then attempt to tear down non-legacy values and vice-versa.

This caused the failures seen [here](https://buildkite.com/rails/rails/builds/76897#cd5c1c84-73de-42b2-b8b7-6084fdd75283)

```
Error:
--

BasePreventWritesTest::BasePreventWritesLegacyTest#test_preventing_writes_with_multiple_handlers:
--
  | NoMethodError: undefined method `[]' for nil:NilClass
  | /rails/activerecord/lib/active_record/test_fixtures.rb:209:in `block (2 levels) in setup_shared_connection_pool'
```

Using the same variable causes the same issue.